manual/tools/gen_invocations.py

```diff
@@ -79,24 +88,37 @@ def gen_invocations(input_files, output_file, args):
         for interface_name, methods in itertools.groupby(methods, lambda x: x[0]):
             group_id = interface_name if prefix is None else prefix + '_' + interface_name
             group_name = interface_name
-            output_file.write("/**\n * @defgroup %s %s\n * @{\n */\n\n" % (group_id, group_name))
-            output_file.write("/** @} */\n")
-            for (interface_name, method_name, method_id, inputs, outputs, _, comment) in methods:
-                prototype = "/**\n * @addtogroup %s %s\n * @{\n */\n\n" % (group_id, group_name)
-                prototype += generate_prototype(interface_name,
-                                                method_name, method_id, inputs, outputs, comment)
-                prototype += "/** @} */\n"
-                prototypes.append(prototype)
 
-        prototypes.sort()
+            if group_id in groups:
+                if group_name != groups[group_id]:
+                    raise Exception(f"Group {group_id} has inconsistent names: {group_name} "
+                                    f"!= {groups[group_id]}")
+            else:
+                groups[group_id] = group_name
 
-        output_file.write("/**\n * @defgroup %s %s\n * @{\n */\n\n" % (api.name, api.name))
-
-        for prototype in prototypes:
-            output_file.write(prototype)
-            output_file.write("\n\n")
-
-        output_file.write("/** @} */\n")
+            for (interface_name, method_name, method_id, inputs, outputs, _, comment) in methods:
+                prototype = "/**\n * @addtogroup %s\n * @{\n */\n\n" % group_id
+                prototype += generate_prototype(interface_name, method_name, method_id, inputs,
+                                                outputs, comment)
+                prototype += "\n/** @} */\n"
+                if group_id not in prototypes:
+                    prototypes[group_id] = [prototype]
+                else:
+                    prototypes[group_id].append(prototype)
+
+        with open(group_file_name, "a") as groups_file:
+            groups_file.write("/**\n * @defgroup %s %s\n * @{\n */\n\n" % (api.name, api.name))
+            for group_id, group_name in sorted(groups.items()):
+                groups_file.write("/**\n * @defgroup %s %s\n */\n\n" % (group_id, group_name))
+            groups_file.write("/** @} */\n\n")
+
+        for group_id, prototypes in prototypes.items():
```

Comment:
I don't like having a shadowed variable here as it is often the source of errors/confusion in Python code, but this is just a minor point.

Reply:
You're right, that is an unfortunate name choice. I'll tweak it.